mpas_analysis/shared/timekeeping/utility.py

```diff
@@ -0,0 +1,229 @@
+"""
+Time keeping utility functions
+
+Author
+------
+Xylar Asay-Davis
+
+Last Modified
+-------------
+02/06/2017
+"""
+
+import datetime
+from .MpasRelativeDelta import MpasRelativeDelta
+
+
+def stringToDatetime(dateString):  # {{{
+    """
+    Given a date string and a calendar, returns a `datetime.datetime`
+
+    Parameters
+    ----------
+    dateString : string
+        A date and time in one of the following formats:
+        - YYYY-MM-DD hh:mm:ss
+        - YYYY-MM-DD hh.mm.ss
+        - YYYY-MM-DD SSSSS
+        - DDD hh:mm:ss
+        - DDD hh.mm.ss
+        - DDD SSSSS
+        - hh.mm.ss
+        - hh:mm:ss
+        - YYYY-MM-DD
+        - YYYY-MM
+        - SSSSS
+
+        Note: either underscores or spaces can be used to separate the date
+        from the time portion of the string.
+
+    Returns
+    -------
+    datetime : A `datetime.datetime` object
+
+    Raises
+    ------
+    ValueError
+        If an invalid `dateString` is supplied.
+
+    Author
+    ------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/04/2017
+    """
+
+    (year, month, day, hour, minute, second) = \
+        _parseDateString(dateString, isInterval=False)
+
+    return datetime.datetime(year=year, month=month, day=day, hour=hour,
+                             minute=minute, second=second)  # }}}
+
+
+def stringToRelativedelta(dateString, calendar='gregorian'):  # {{{
+    """
+    Given a date string and a calendar, returns an instance of
+    `MpasRelativeDelta`
+
+    Parameters
+    ----------
+    dateString : string
+        A date and time in one of the following formats:
+        - YYYY-MM-DD hh:mm:ss
+        - YYYY-MM-DD hh.mm.ss
+        - YYYY-MM-DD SSSSS
+        - DDD hh:mm:ss
+        - DDD hh.mm.ss
+        - DDD SSSSS
+        - hh.mm.ss
+        - hh:mm:ss
+        - YYYY-MM-DD
+        - YYYY-MM
+        - SSSSS
+
+        Note: either underscores or spaces can be used to separate the date
+        from the time portion of the string.
+
+    calendar: {'gregorian', 'gregorian_noleap'}, optional
+        The name of one of the calendars supported by MPAS cores
+
+    Returns
+    -------
+    relativedelta : An `MpasRelativeDelta` object
+
+    Raises
+    ------
+    ValueError
+        If an invalid `dateString` is supplied.
+
+    Author
+    ------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/04/2017
+    """
+
+    (years, months, days, hours, minutes, seconds) = \
+        _parseDateString(dateString, isInterval=True)
+
+    return MpasRelativeDelta(years=years, months=months, days=days,
+                             hours=hours, minutes=minutes, seconds=seconds,
+                             calendar=calendar)
+    # }}}
+
+
+def clampToNumpyDatetime64(date, yearOffset):
+    """
+    Temporary function for adding an offset year and clamping a datetime to
+    range supported by `numpy.datetime64`.
+    """
```

Comment:
If we are going to get rid of this on a short timescale of weeks, but not months, this is noise and we should proceed in the less rigorous way.